app/main.py

```diff
@@ -1,34 +1,65 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.start = Deck(start[0], start[1])
+        self.end = Deck(end[0], end[1])
+        self.is_drowned = is_drowned
+        diff_row = self.end.row - self.start.row
+        diff_column = self.end.column - self.start.column
+        self.decks = [self.start]
+        if diff_row != 0:
+            for i in range(diff_row):
+                new_deck = Deck(self.start.row + (i + 1), self.start.column)
+                self.decks.append(new_deck)
+        if diff_column != 0:
+            for i in range(diff_column):
+                new_deck = Deck(self.start.row, self.start.column + (i + 1))
+                self.decks.append(new_deck)
```

Comment:
you can handle it in two nested cycles

Reply:
by the way, I would pull off this logic in the method, instead of writing everything in a constructor